include/inviwo/core/datastructures/histogram.h

```diff
@@ -33,96 +33,70 @@
 #include <vector>
 #include <bitset>
 #include <array>
+#include <span>
 
 namespace inviwo {
-enum class HistogramMode { Off, All, P99, P95, P90, Log };
+enum class HistogramMode : int { Off = 0, All, P99, P95, P90, Log };
+constexpr size_t numberOfHistogramModes = 6;
 
 using HistogramSelection = std::bitset<32>;
 constexpr HistogramSelection histogramSelectionAll{0xffffffff};
 
-/**
- *The NormalizedHistogram has a array of bins and all bins are normalized.
- *It can be de-normalized using the maxiumBinValue_.
- */
-class IVW_CORE_API NormalizedHistogram {
-public:
-    struct Stats {
-        double min;
-        double max;
-        double mean;
-        double standardDeviation;
-        std::vector<double> percentiles;
-    };
-
-    NormalizedHistogram();
-
-    NormalizedHistogram(dvec2 dataRange, std::vector<double> counts, double min, double max,
-                        double mean, double standardDeviation);
-
-    std::vector<double>& getData();
-    const std::vector<double>& getData() const;
-    double& operator[](size_t i);
-    const double& operator[](size_t i) const;
-
-    double getMaximumBinValue() const;
-
-    Stats stats_;
-    Stats histStats_;
-    dvec2 dataRange_;
-
-protected:
-    std::vector<double> data_;
-    double maximumBinCount_;
+struct IVW_CORE_API Statistics {
+    double min;
+    double max;
+    double mean;
+    double standardDeviation;
+    std::vector<double> percentiles;
 };
 
-class IVW_CORE_API HistogramContainer {
-public:
-    HistogramContainer() = default;
-    template <typename FirstIter, typename LastIter>
-    HistogramContainer(dvec2 range, size_t bins, FirstIter begin, LastIter end);
+struct IVW_CORE_API Histogram1D {
+    std::vector<size_t> counts;
+    size_t totalCounts{0};
+    size_t maxCount{0};
+    DataMapper dataMap{};
+    size_t underflow{0};
+    size_t overflow{0};
 
-    const NormalizedHistogram& operator[](size_t i) const;
-    const NormalizedHistogram& get(size_t i) const;
-
-    NormalizedHistogram& operator[](size_t i);
-    NormalizedHistogram& get(size_t i);
-
-    size_t size() const;
-    bool empty() const;
-
-    void clear();
-
-private:
-    std::vector<NormalizedHistogram> histograms_;
+    Statistics dataStats;
+    Statistics histStats;
 };
 
-template <typename FirstIter, typename LastIter>
-HistogramContainer::HistogramContainer(dvec2 dataRange, size_t bins, FirstIter begin,
-                                       LastIter end) {
-    using T = typename std::iterator_traits<FirstIter>::value_type;
+namespace util {
+
+IVW_CORE_API std::vector<double> calculatePercentiles(const std::vector<size_t> hist, dvec2 range,
+                                                      size_t sum);
+IVW_CORE_API Statistics calculateHistogramStats(const std::vector<size_t>& hist);
 
+template <typename T>
+std::vector<Histogram1D> calculateHistograms(std::span<const T> data, const DataMapper& dataMap,
+                                             size_t bins) {
```

Comment:
with this change we can change hists below to bins

```suggestion
                                             size_t numBins) {
```

Reply:
that does not work, since hists here represent multiple histograms.